tests/conftest.py

```diff
@@ -5,8 +5,22 @@
 
 import pytest
 
-from starlette.testclient import TestClient
-from tests.types import TestClientFactory
+from starlette.testclient import AsyncTestClient, TestClient
+from tests.types import AsyncTestClientFactory, TestClientFactory
+
+
+@pytest.fixture
+def async_test_client_factory(
+    anyio_backend_name: Literal["asyncio", "trio"],
+    anyio_backend_options: dict[str, Any],
+) -> AsyncTestClientFactory:
+    # anyio_backend_name defined by:
+    # https://anyio.readthedocs.io/en/stable/testing.html#specifying-the-backends-to-run-on
+    return functools.partial(
+        AsyncTestClient,
+        backend=anyio_backend_name,
```

Comment:
Why would AsyncTestClient need the backend? Surely it would sniffio it from the first async call?